RFC/src/RFC-0230_HTLC.md

```diff
@@ -48,49 +48,75 @@ technological merits of the potential system outlined herein.
+
+### Mimblewimble N-of-N Multisig UTXO
+
+A normal Mimblewimble UTXO does not have a notion of being a [multisig] UTXO. The UTXO is hidden inside the commitment `C(v,r) = r·G + v·H` by virtue of being blinded. However, the blinding factor `r` can be composed of multiple blinding factors where `r = r1 + r2 + ... + rn`. The output commitment can then be constructed as `C(v,r) = r1·G + r2·G + ... + rn·G + v·H = (r1 + r2 + ... + rn)·G + v·H` where each participant keeps their private blinding factor hidden and only provides their public blinding factor. A multi-party aggregated signature (aggsig) scheme like Musig may be employed to sign such a transaction so that all parties' interests can be protected.
+
+The base layer is oblivious as to how the commitment and related signature were constructed. To open such commitments (in order to spend it) only the n-of-n blinding factor `r` is required, and not the original aggsig that was used to sign the transaction. The parties that wants to open the commitment needs to collaborate with special scriptless scripts to produce the n-of-n blinding factor `r`.
+
+### Hashed Time Locked Contract
+
+Unlike Bitcoin where an [HTLC] can be accomplished with a single transaction, in [Mimblewimble] it is a multi-step process. 
+
+The steps where only one sender and one receiver are involved are as follows:
+* The sender MUST pay all the funds into a [2-of-2 multisig UTXO](#mimblewimble-n-of-n-multisig-utxo), where the participants are both the sender and receiver.
+* The parties involved MUST construct a refund [transaction] paying back all funds to the sender from the 2-of-2 multisig [UTXO]. This transaction MUST have a [transaction lock height](#time-locked-transactions) in the far future so that it cannot be mined immediately. It therefor lives in the [mempool].
+* The parties involved MUST publish both above [transaction]s at the same time. 
+* The parties involved MUST construct a payout [transaction] that pay the receiver the funds. This type of [transaction] typically makes use of a preimage (similar to [adapter signatures](introduction-to-scriptless-scripts.html#adaptor-signatures)) to allow spending of the [transaction] if the user reveals some knowledge, allowing the user to unlock the UTXO.
+
+HTLC's in [Mimblewimble] makes use of double spending the n-of-n [multisig] UTXO and the first valid published [transaction] can then be mined and claim the n-of-n multisig UTXO. 
+
+An example of an [HTLC] in practice can be viewed at Tari University:
+[Mimblewimble atomic swaps](https://tlu.tarilabs.com/protocols/grin-protocol-overview/MainReport.html#atomic-swaps)
+
+[HTLC]: Glossary.md#hashed-time-locked-contract
 [mempool]: Glossary.md#mempool
-[mimblewimble]: Glossary.md#mimblewimble
+[Mimblewimble]: Glossary.md#mimblewimble
 [base node]: Glossary.md#base-node
 [block]: Glossary.md#block
-[currenthead]: Glossary.md#currenthead
+[current head]: Glossary.md#current-head
```

Comment:
I think the topic is flagged as currenthead

Reply:
I checked, it is `https://rfc.tari.com/Glossary.html#current-head`, thus we need the dash in the link.

Reply:
cwl will update then